tests/profile/test_user_configurable_profiler_v3_batch_request.py

```diff
@@ -0,0 +1,927 @@
+    titanic_data_context_modular_api, taxi_validator_spark, possible_expectations_set
+):
+    """
+    What does this test do and why?
+    Ensures that all available expectation types work as expected for spark
+    """
+    if taxi_validator_spark is not None:
+        context = titanic_data_context_modular_api
+
+        ignored_columns = [
+            "pickup_location_id",
+            "dropoff_location_id",
+            "fare_amount",
+            "extra",
+            "mta_tax",
+            "tip_amount",
+            "tolls_amount",
+            "improvement_surcharge",
+            "congestion_surcharge",
+        ]
+
+        semantic_types = {
+            "datetime": ["pickup_datetime", "dropoff_datetime"],
+            "numeric": ["total_amount", "passenger_count"],
+            "value_set": [
+                "payment_type",
+                "rate_code_id",
+                "store_and_fwd_flag",
+                "passenger_count",
+            ],
+            "boolean": ["store_and_fwd_flag"],
+        }
+
+        profiler = UserConfigurableProfiler(
+            taxi_validator_spark,
+            semantic_types_dict=semantic_types,
+            ignored_columns=ignored_columns,
+            # TODO: Add primary_or_compound_key test
+            #  primary_or_compound_key=[
+            #     "vendor_id",
+            #     "pickup_datetime",
+            #     "dropoff_datetime",
+            #     "trip_distance",
+            #     "pickup_location_id",
+            #     "dropoff_location_id",
+            #  ],
+        )
+
+        assert profiler.column_info.get("rate_code_id")
+        suite = profiler.build_suite()
+        assert len(suite.expectations) == 45
+        (
+            columns_with_expectations,
+            expectations_from_suite,
+        ) = get_set_of_columns_and_expectations_from_suite(suite)
+
+        unexpected_expectations = {
+            "expect_column_values_to_be_unique",
+            "expect_column_values_to_be_null",
+            "expect_compound_columns_to_be_unique",
+        }
+        assert expectations_from_suite == {
+            i for i in possible_expectations_set if i not in unexpected_expectations
+        }
+
+        ignored_included_columns_overlap = [
+            i for i in columns_with_expectations if i in ignored_columns
+        ]
+        assert len(ignored_included_columns_overlap) == 0
+
+        results = context.run_validation_operator(
+            "action_list_operator", assets_to_validate=[taxi_validator_spark]
+        )
+
+        assert results["success"]
+
+
+def test_profiler_all_expectation_types_sqlalchemy(
+    titanic_data_context_modular_api,
+    taxi_validator_sqlalchemy,
+    possible_expectations_set,
+):
+    """
+    What does this test do and why?
+    Ensures that all available expectation types work as expected for spark
+    """
+    if postgresqltypes is not None and taxi_validator_sqlalchemy is not None:
```

Comment:
To handle import/dependency issues in tests, I think we want to follow one of the following patterns:

```
# as a decorator
@pytest.mark.skipif(
    importlib.util.find_spec("pyspark") is None, reason="requires the Spark library"
)
@pytest.fixture
def test_dataframe(spark_session):
```
or
```
# within an if statement
def test_instantiation_via_credentials(sa, test_backends, test_df):
    if "postgresql" not in test_backends:
        pytest.skip("test_database_store_backend_get_url_for_key requires postgresql")
```
or
```
# using pytest.importorskip
try:
    sqlalchemy = pytest.importorskip("sqlalchemy")
except ImportError:
    sqlalchemy = None
```

I believe if you use `pytest.importorskip` at the module level, all tests in module are skipped, vs. calling at the beginning of a test function.

Reply:
That way we know that something was deliberately skipped because of a missing dependency, vs. silently not executing.